server/src/db/repositories/recipe.repository.ts

```diff
@@ -0,0 +1,47 @@
+import { ErrorUtils } from "./../../logic/utils/index";
+import { DbError, ERRORS, Recipe } from "./../../models/index";
+
+export class RecipeRepository {
+
+  private _GET_BY_ID: string = `select * 
+                                from recipes
+                                where recipes.id = $1`;
+
+  private _SAVE: string = `insert into recipes(
+                             title, userId, description,
+                             imageUrl, created)
+                           values($1, $2, $3, $4, $5)
+                           returning id`;
+
+  constructor (private _db: any) {}
```

Comment:
after this change we have no intellisense for the database methods, but we can't do this:

```typescript
constructor (private _db: IDatabase<any>) {}
```

we need to do this:

```typescript
private _db: IDatabase<any>;

constructor (db: any) {
  this._db = db;
}
```

Should we change this?